online-workflow/src/main/java/eu/itesla_project/online/db/OnlineDbMVStore.java

```diff
@@ -1087,32 +1103,65 @@ public int compare(Integer o1, Integer o2) {
     }
 
     @Override
-    public Network getState(String workflowId, Integer stateId) {
+    public Map<Integer, Set<String>> listStoredPostContingencyStates(String workflowId) {
+        LOGGER.info("Getting list of stored post contingency states for workflow {}", workflowId);
+        Map<Integer, Set<String>> storedContingencyStates = new TreeMap<>();
+        if (workflowStatesFolderExists(workflowId)) {
+            Path workflowStatesFolder = getWorkflowStatesFolder(workflowId);
+            File[] files = workflowStatesFolder.toFile().listFiles(new FilenameFilter() {
+                public boolean accept(File dir, String name) {
+                    return name.toLowerCase().startsWith(STORED_STATE_POST_PREFIX);
+                }
+            });
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    List<String> stateIdContId = Splitter.on(STORED_STATE_CONT_PREFIX).omitEmptyStrings().trimResults().splitToList(file.getName().substring(STORED_STATE_POST_PREFIX.length()));
+                    Integer stateId = Integer.parseInt(stateIdContId.get(0));
+                    String contId = stateIdContId.get(1);
+                    if (storedContingencyStates.containsKey(stateId)) {
+                        storedContingencyStates.get(stateId).add(contId);
+                    } else {
+                        storedContingencyStates.put(stateId, new HashSet<String>(Arrays.asList(contId)));
+                    }
+                }
+            }
```

Comment:
This block should use the Java NIO API (Files.walk) to be independant of the file storage implementation (local HDD, HDFS...)